sound/Music.java

```diff
@@ -187,37 +192,91 @@ public void playChord(Chord chord, int defaultDuration) {
             System.out.println("Problem sleeping?");
         }
     }
-    
+
     /**
      * Play a tune.
-     * @param tune the tune to play
+     * @param tune the tune to play.
+     * @param imp the ImgProvider object.
      */
-    public void playTune(Tune tune) {
+    public void playTune(final Tune tune, final ImgProvider imp) {
+        ImgProvider improvider = imp;
+        int[] pixels = improvider.getPix();
+        int height = improvider.getPixHeight();
+        int width = improvider.getPixWidth();
+        int newRow = width;
+	/**Holds increasing amounts of pixels
+	 *that are used to display more of the
+	 *image for each chord played.*/
+        int[] row = new int[width * height];
         /** Current chord */
         Chord chord;
-        /** Current duration */
+        /**Current duration */
         int duration = STD_DURATION;
-        /** Tune iterator */
+        /**Tune iterator */
         Iterator<Chord> tuneIt = tune.iterator();
         int line = 0;
         while (tuneIt.hasNext()) {
+            for (int i = 0; i < newRow; i++) {
+                row[i] = pixels[i];
+            }
+            improvider.getDis().synchronize(width, height, row);
+            newRow += width;
             System.out.println("[Line " + (line++) + "] ");
             chord = tuneIt.next();
-            playChord(chord,duration);
+            playChord(chord, duration);
         }
         silence();
     }
-    
+
     /**
      * Stop all tones coming from synthesizer.
      */
     public void silence() {
         Note note;
         int numNotes = noteHistory.size();
-        for (int i=0; i < numNotes; i++) {
+        for (int i = 0; i < numNotes; i++) {
             note = noteHistory.get(i);
-            channels[note.channel()].noteOff(note.pitch(),note.velocityOn());
+            channels[note.channel()].noteOff(note.pitch(), note.velocityOn());
         }
     }
 
+    /**
+     *Accessor for numChannels.
+     *@return numChannels.
+     */
+    public int getNumChannels() {
+        return numChannels;
+    }
+
+    /**
+     *Accessor for synth.
+     *@return synth.
+     */
+    public Synthesizer getSynthesizer() {
+        return synth;
+    }
+
+    /**
+     *Accessor for channel.
+     *@return channels.
+     */
+    public MidiChannel[] getMidiChannel() {
+        return channels;
+    }
+
+    /**
+     *Accessor for instrument.
+     *@return instrument.
+     */
+    public int getInstrument() {
+        return instrument;
+    }
+
+    /**
+     *Accessor for velocity.
+     *@return velocity.
+     */
+    public int getVelocity() {
+        return velocity;
+    }
```

Comment:
Javadoc formatting.

Additional Note: The @return tag should specify the _meaning_ of what is returned.  It also should end with a period only if it is a complete sentence.  In these cases it seems that the method names and the @return tags are exposing the internals (private field names) rather than providing a meaningful API.  Where are these "getter" methods used that justifies the public API?